listenbrainz/webserver/views/profile.py

```diff
@@ -241,6 +241,36 @@ def delete():
             user=current_user,
         )
 
+@profile_bp.route('/delete-listens', methods=['GET', 'POST'])
+@login_required
+def delete_listens():
+    """ Delete all the listens for the currently logged-in user from ListenBrainz.
+
+    If POST request, this view checks for the correct authorization token and
+    deletes the listens. If deletion is successful, redirects to user's profile page, 
+    else flashes an error and redirects to user's info page.
+
+    If GET request, this view renders a page asking the user to confirm
+    that they wish to delete their listens.
+    """
+    if request.method == 'POST':
+        if request.form.get('token') == current_user.auth_token:
+            try:
+                delete_listens_history(current_user.musicbrainz_id)
+            except Exception as e:
+                current_app.logger.error('Error while deleting listens for %s: %s', current_user.musicbrainz_id, str(e))
+                flash.error('Error while deleting listens for %s, please try again later.' % current_user.musicbrainz_id)
+                return redirect(url_for('profile.info'))
+            flash.info('Successfully deleted listens for %s.' % current_user.musicbrainz_id)
+            return redirect(url_for('user.profile', user_name=current_user.musicbrainz_id))
+        else:
+            flash.error('Cannot delete the user\'s listens due to error during authentication, please try again later.')
```

Comment:
This message is shown to the user. Let's change it to something more user friendly.

"We couldn't delete your listens due to an error. Please try again later."

Reply:
I have updated it with ### Auth token invalid or missing.